functional/io.py

```diff
@@ -290,14 +285,12 @@ def read(self):
 def get_read_function(filename: str, disable_compression: bool):
     if disable_compression:
         return ReusableFile
-    else:
-        with open(filename, "rb") as f:
-            start_bytes = f.read(N_COMPRESSION_CHECK_BYTES)
-            for cls in COMPRESSION_CLASSES:
-                if cls.is_compressed(start_bytes):
-                    return cls
-
-            return ReusableFile
+    with open(filename, "rb") as f:
+        start_bytes = f.read(N_COMPRESSION_CHECK_BYTES)
+        return next(
+            (cls for cls in COMPRESSION_CLASSES if cls.is_compressed(start_bytes)),
```

Comment:
Any reason for change of style? I think its harder to read, but could accept it if there is an advantage

Reply:
reverted